### PR TITLE
Add initial Archipelago support

### PR DESCRIPTION
**Pull request by decentralion**
This is an initial add of Archipelago support, currently it will detect listings on Archipelago and put them in the listings channel. I will add support for trades in a followon PR. 

Tested on the artbot-testing discord, produced embeds like this:
image

### CODE REVIEW

**Review by grantoesterling — APPROVED** [review]

File: Classes/ArchipelagoBot.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        JSON.stringify({ type: 'SUBSCRIBE_TOPIC', topic: 'ALL_COLLECTIONS' })
+      )
+    this.client.on('open', () => subscribe())
+    this.client.on('message', (message) => this.onMessage(message))
+  }
+
+  async refreshCollections() {
+    this.slugToCollections = new Map()
+    const collectionsResponse = await fetch(
+      'https://api.archipelago.art/v1/market/collections'
```
[the remark this answers is not in the document]
Comment (jakerockland) [reply Classes/ArchipelagoBot.js:33]: ++ 🙏 😄

File: Classes/ArchipelagoBot.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    }
+    return collection
+  }
+
+  async getArtBlocksData(slug, tokenIndex) {
+    const collection = await this.getCollection(slug)
+    if (collection == null) {
+      return null
+    }
+    const offset = collection.artblocksProjectIndex * 1000000
```
Comment (jakerockland) [inline Classes/ArchipelagoBot.js:58]: Nit: can 1000000 be a top-of-file `const`?

**Review by jakerockland — APPROVED** [review]

File: Classes/ArchipelagoBot.js (hunk 2 above)
Comment (grantoesterling) [reply Classes/ArchipelagoBot.js:58]: yeah call it `ONE_MILLION`

**Comment by decentralion** [comment]
Addressed nits :)

**Review by jakerockland — APPROVED** [review]
Awesome–TYVM for the updates @decentralion 

Merging!

**Pull request merged**